packages/app/src/cli/models/app/loader.test.ts

```diff
@@ -1664,8 +1671,42 @@ automatically_update_urls_on_dev = true
     })
   })
 
+  test('throws error if config file is passed in but does not exist', async () => {
+    // Given
+    await writeConfig(linkedAppConfiguration)
+    vi.mocked(getCachedAppInfo).mockReturnValue({directory: tmpDir, configFile: 'shopify.app.non-existent.toml'})
+    vi.mocked(use).mockResolvedValue('shopify.app.toml')
+
+    // When
+    const result = loadApp({directory: tmpDir, specifications, configName: 'non-existent'})
+
+    // Then
+    await expect(result).rejects.toThrow(`Couldn't find shopify.app.non-existent.toml in ${tmpDir}.`)
+    expect(use).not.toHaveBeenCalled()
+  })
+
   const runningOnWindows = platformAndArch().platform === 'windows'
 
+  test.skipIf(runningOnWindows)(
```

Comment:
What's the problem with Windows here?

Reply:
I think this test should really be happening on Windows -- the code will be run there after all and this is really fundamental functionality? `normalizePath` should cover this IIRC

Reply:
All tests (including Windows) passing now